src/Kralizek.Extensions.Configuration.AWSSecretsManager/Internal/SecretsManagerConfigurationProvider.cs

```diff
@@ -139,17 +139,23 @@ void SetData(IEnumerable<(string, string)> values, bool triggerReload)
             }
         }
 
-        async Task<IReadOnlyList<SecretListEntry>> FetchAllSecretsAsync(CancellationToken cancellationToken)
+        private async Task<IReadOnlyList<SecretListEntry>> FetchAllSecretsAsync(CancellationToken cancellationToken)
         {
             var response = default(ListSecretsResponse);
 
             var result = new List<SecretListEntry>();
 
+            if (Options.UseDefinedArnList)
```

Comment:
Check if the list contains items instead of using the flag

Reply:
`.Count > 0` Good idea, done.